examples/server.rs

```diff
@@ -113,31 +90,52 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
                         .unwrap();
 
                     while let Some((req, stream)) = h3_conn.accept().await.unwrap() {
-                        debug!("connection requested: {:#?}", req);
-
-                        tokio::spawn(handle_request(stream));
+                        let root = root.clone();
+                        debug!("New request: {:#?}", req);
+
+                        tokio::spawn(async {
+                            if let Err(e) = handle_request(req, stream, root).await {
+                                error!("request failed: {}", e);
+                            }
+                        });
                     }
                 }
                 Err(err) => {
-                    warn!("connecting client failed with error: {:?}", err);
+                    warn!("accepting connection failed: {:?}", err);
                 }
             }
         });
     }
 
+    endpoint.wait_idle().await;
+
     Ok(())
 }
 
 async fn handle_request<T>(
+    req: Request<()>,
     mut stream: RequestStream<T, Bytes>,
-) -> Result<(), Box<dyn std::error::Error + Send>>
+    serve_root: Arc<Option<PathBuf>>,
+) -> Result<(), Box<dyn std::error::Error>>
 where
     T: BidiStream<Bytes>,
 {
-    let resp = http::Response::builder()
-        .status(http::StatusCode::NOT_FOUND)
-        .body(())
-        .unwrap();
+    let (status, to_serve) = match serve_root.as_deref() {
+        None => (StatusCode::IM_A_TEAPOT, None),
```

Comment:
Does a browser or client know what to do with such a response?

Reply:
I don't think so, I felt like putting a joke in there when miss-configured (root directory is missing). Unfortunately, browsers don't get the humor: 

![image](https://user-images.githubusercontent.com/6791052/148516550-25854b5c-7b8a-4eb3-992b-6746246082c1.png)

I didn't want to automatically open the current directory to the world, so I made the argument optional and said that the server had a teapot feature instead.

If you want to get it removed, I can either make the parameter mandatory or set it to a mostly sane default, like `www` for example.

Reply:
I agree it shouldn't default to the current directory. Since this is an example, I think it'd be good if a simple "hello world" response were sent with a 200 status.